sdk/lib/opentelemetry/sdk/trace/span.rb

```diff
@@ -331,6 +331,8 @@ def trim_links(links, link_count_limit, attribute_per_link_count_limit) # ruboco
           # Fast path (likely) common cases.
           return nil if links.nil?
 
+          links = links.map { |link| link.span_context.valid? }
```

Comment:
This turns all the links into booleans, which isn't what we want.
```suggestion
          links = links.select { |link| link.span_context.valid? }
```
Is correct, but inefficient. We have a fast path below that should also check for validity.
```suggestion
          links = links.map { |link| link.span_context.valid? }
```

Reply:
Did you mean this?

    link.attributes.size <= attribute_per_link_count_limit && Internal.valid_attributes?(name, 'link', link.attributes)